### PR TITLE
Better Ollama support

### PR DESCRIPTION
**Pull request by dillonalaird**

### CODE REVIEW

File: vision_agent/agent/vision_agent_coder.py [hunk 1]
```diff
@@ -849,6 +854,64 @@ def log_progress(self, data: Dict[str, Any]) -> None:
...
+    Example
+    -------
+        >>> image vision_agent as va
+        >>> agent = va.agent.OllamaVisionAgentCoder()
+        >>> code = agent("What percentage of the area of the jar is filled with coffee beans?", media="jar.jpg")
+    """
+
+    def __init__(
+        self,
+        planner: Optional[LMM] = None,
```
[the remark this answers is not in the document]
Comment (dillonalaird) [reply vision_agent/agent/vision_agent_coder.py:876]: You can mix and match any LMM/LLM here

File: vision_agent/lmm/lmm.py [hunk 2]
```diff
@@ -330,12 +330,16 @@ def __init__(
         model_name: str = "llava",
         base_url: Optional[str] = "http://localhost:11434/api",
         json_mode: bool = False,
+        num_ctx: int = 128_000,
```
[the remark this answers is not in the document]
Comment (dillonalaird) [reply vision_agent/lmm/lmm.py:333]: yes let me add a link to the ollama doc, it's not super obvious what this is https://github.com/ollama/ollama/blob/main/docs/modelfile.md#valid-parameters-and-values

File: vision_agent/lmm/lmm.py [hunk 3]
```diff
@@ -408,21 +414,22 @@ def generate(
...
+        data: Dict[str, Any] = {
             "model": self.model_name,
             "prompt": prompt,
             "images": [],
         }
 
         if media and len(media) > 0:
             for m in media:
-                data["images"].append(encode_media(m))  # type: ignore
+                data["images"].append(encode_media(m))
```
[the remark this answers is not in the document]
Comment (dillonalaird) [reply vision_agent/lmm/lmm.py:425]: It cannot, but you can still use it. I added a comment in the README and the class documentation, but llava actually errors out with a 5xx on these longer prompts so it's not possible to use it. Llama 3.1 will work but definitely the smaller models struggle. I've added a warning about this in the README as well.

File: vision_agent/utils/sim.py [hunk 4]
```diff
@@ -105,17 +111,51 @@ def __init__(
...
+    def __init__(
+        self,
+        df: pd.DataFrame,
+        sim_key: Optional[str] = None,
+        model_name: Optional[str] = None,
+        base_url: Optional[str] = None,
+    ) -> None:
+        self.df = df
+        if base_url is None:
+            base_url = "http://localhost:11434/api/embeddings"
```
[the remark this answers is not in the document]
Comment (dillonalaird) [reply vision_agent/utils/sim.py:141]: This is the default URL for ollama

**Review by shankar-vision-eng — APPROVED** [review]
LGTM! 

P.S. Approving the even though integration tests are failing probably because end point is down. This PR doesn't touch the tools

**Pull request merged**